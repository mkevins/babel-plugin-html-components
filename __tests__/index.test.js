const babel = require('@babel/core');
const plugin = require('../src/');

describe('plugin', () => {
  it('transforms div to Div', () => {
    let source = `
      class MyComponent extends Component {
        render() {
          return <div className={ 'styled' } />;
        }
      }
    `;

    const {code} = babel.transform(source, {plugins: [
      '@babel/plugin-syntax-jsx',
      plugin
    ]});

    expect(code).toMatchSnapshot();
  });
});

