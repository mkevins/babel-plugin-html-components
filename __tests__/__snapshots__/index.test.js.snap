// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`plugin transforms div to Div 1`] = `
"class MyComponent extends Component {
  render() {
    return <Div className={'styled'} />;
  }

}"
`;
